### ISSUE TITLE
Exceptions are swallowed whole

### ISSUE THREAD

**Issue by colinnewell** [dagolden/Try-Tiny-Retry#2]
I expected the exception that was not caught as part of the retry process to be re-thrown.  I'm not sure whether this a documentation or code issue.

I expected this code,

```
retry
{
    ...
}
retry_if {/current transaction is aborted, commands ignored until end of transaction block/};
```

To then throw the exception it failed to catch (either too many transactions aborted, or an unexpected exception type).

Instead I needed,

```
retry
{
    ...
}
retry_if {/current transaction is aborted, commands ignored until end of transaction block/}
catch
{
    die $_;
};
```

**Comment by dagolden** [comment]
That's a documentation issue.  If the retry_if clause fails, then it's just like using try BLOCK with no catch.

I'll figure out a way to clarify that in the docs.  Thanks for pointing it out.

**Comment by dagolden** [comment]
Is this clear enough: 666b183e239c9b4b0b14748b6b03576dfd492fa3

**Comment by colinnewell** [comment]
That's almost correct.  It will also re-throw the exception mentioned in
the retry_if if you've hit the maximum retries.

**Comment by dagolden** [comment]
That might be a bug.  Without a catch block, the maximum retries should just eat the error.  (Well, set it into $@ anyway).

**Comment by colinnewell** [comment]
Actually I meant it was just an issue with the way the sentence reads.  The
way I read that sentence it appears that you're saying that the catch will
only re-throw the exceptions not mentioned by the 'retry_if' blocks.  It
also re-throws the exceptions mentioned in the retry_if blocks assuming
that the retry limit is hit.

You are correct that it eats it if there is no catch statement which is
consistent.

The behaviour seems logical and correct.

**Comment by dagolden** [comment]
I've changed the docs further and released 0.003 to CPAN.
